From a8ff80a80dc025e223984dd94584e95bb4effb5e Mon Sep 17 00:00:00 2001
From: Chuansheng Liu <chuansheng.liu@intel.com>
Date: Tue, 10 Feb 2015 09:44:30 +0800
Subject: [PATCH 131/441] ISH: fix the memory leak for dev_name

After finished the using of dev_name string, we need to free it immediately,
Otherwise memory leak is there.

Change-Id: I6d3dc245ae3148bbbd0f1f2c718a61671b89dd61
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-21566
Signed-off-by: Chuansheng Liu <chuansheng.liu@intel.com>
(cherry picked from commit ced4437d70d2f74c31701bb3f0106cbbced93d47)
---
 drivers/misc/heci/bus.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/misc/heci/bus.c
+++ b/drivers/misc/heci/bus.c
@@ -574,6 +574,7 @@ int	heci_bus_new_client(struct heci_devi
 	device_create_file(&cl_device->dev, &dma_hdr_len);
 	device_create_file(&cl_device->dev, &num_active_connections);
 	device_create_file(&cl_device->dev, &client_id);
+	kfree(dev_name);
 
 	return	0;
 }
